docs/current/architecture.md

```diff
@@ -22,68 +22,74 @@
+1. `Event API`: REST API that receives events from long running services on agents.
+1. `Server Controller`: Responsible for maintaining all services that Vaquero Server provides, while acting as an intermediary between the services.
+1. `State Engine`: The brains behind vaquero; understands the data model, looks at event and task history, and ultimately makes the decision on moving provisioning hosts from one state to the next.
+1. `Task Manager`: Responsible for running jobs on distributed task executors; e.g, running a container to execute a power management call to reboot a host for reprovisioning.
 
-1. **user API (in progress)** - REST API for users to interact with the system and provide operational insights into booting hosts.
-2. **updater** - The interface that SoT's will go through to update their data models, could be listening for github webhooks or watching a source of truth directory for modifications.
-3. **model API** - REST API that responds to agents looking to update their model cache, it will also provide a state manifest that enables an agent to know what state its booting hosts are in.
-4. **event API** - REST API that receives events from long running services on agents.
-5. **server controller** - The process that manages the numerous go routines on vaquero servers and acts as an intermediary between all server services.
-6. **state engine** - The brains behind vaquero; will understand a data model, look at the events history and task history to understand what actions need to be taken to move booting hosts from one state to the next.
-7. **task manager** - A generic container task manager that knows how to run jobs on distributed task executors. A use case will be to run LOM containers that will execute on an agent to reboot a host, kicking off the re-provision process.
 
+### Vaquero Agent
 
-#### `vaquero agent`
+A Vaquero agent is responsible for the provisioning of a local datacenter. Many Vaquero agents can be used with a single Vaquero server. Agents are stateless by design and can be created and destroyed as needed. Agents obtain state, configuration and actions by checking in with the Vaquero server. Agents consist of the following components:
 
-The `vaquero` container running in `agent` mode registers itself with an upstream vaquero server and drives provisioning in a local datacenter. The agent is designed in such a way that it is stateless, agents can be created and destroyed at will. They will obtain state, configuration, actions from the vaquero server cluster.
+1. `Model Cache`: Handles just-in-time updates from vaquero server to serve as a cached representation of the data model; can also be used to cotinue operations during network outages.
+1. `Agent Controller`: Responsible for maintaining all services that Vaquero Agent provides, while acting as an intermediary between the services.
```

Comment:
`that` awkward `the Vaquero Agent provides`

Reply:
`Acts as an intermediary between all of the services that the agent controller provides.`